model_compression_toolkit/data_generation/pytorch/optimization_functions/lr_scheduler.py

```diff
@@ -0,0 +1,225 @@
+    Reduce learning rate when a metric has stopped improving. This scheduler allows resetting
+    the learning rate to the initial value after a specified number of bad epochs.
+    """
+
+    def __init__(self, optimizer: Optimizer, mode: str = 'min', factor: float = 0.1, patience: int = 10,
+                 threshold: float = 1e-4, threshold_mode: str = 'rel', cooldown: int = 0,
+                 min_lr: Union[float, List[float]] = 0, eps: float = 1e-8, verbose: bool = False):
+        """
+        Initialize the ReduceLROnPlateauWithReset scheduler.
+
+        Args:
+            optimizer (Optimizer): Wrapped optimizer.
+            mode (str): One of `min`, `max`. In `min` mode, lr will be reduced when the quantity
+                        monitored has stopped decreasing; in `max` mode it will be reduced when the
+                        quantity monitored has stopped increasing. Default: 'min'.
+            factor (float): Factor by which the learning rate will be reduced. new_lr = lr * factor.
+                            Default: 0.1.
+            patience (int): Number of epochs with no improvement after which learning rate will be reduced.
+                            Default: 10.
+            threshold (float): Threshold for measuring the new optimum, to only focus on significant changes.
+                               Default: 1e-4.
+            threshold_mode (str): One of `rel`, `abs`. In `rel` mode, dynamic_threshold = best * ( 1 + threshold )
+                                  in 'max' mode or best * ( 1 - threshold ) in `min` mode. In `abs` mode, dynamic_threshold
+                                  = best + threshold in `max` mode or best - threshold in `min` mode. Default: 'rel'.
+            cooldown (int): Number of epochs to wait before resuming normal operation after lr has been reduced.
+                            Default: 0.
+            min_lr (float or list): A scalar or a list of scalars. A lower bound on the learning rate of all param groups
+                                    or each group respectively. Default: 0.
+            eps (float): Minimal decay applied to lr. If the difference between new and old lr is smaller than eps,
+                         the update is ignored. Default: 1e-8.
+            verbose (bool): If True, prints a message to stdout for each update. Default: False.
+        """
+        if factor >= 1.0:
+            raise ValueError('Factor should be < 1.0.')
+        self.factor = factor
+
+        # Attach optimizer
+        if not isinstance(optimizer, Optimizer):
+            raise TypeError('{} is not an Optimizer'.format(
+                type(optimizer).__name__))
+        self.optimizer = optimizer
+
+        if isinstance(min_lr, (list, tuple)):
+            if len(min_lr) != len(optimizer.param_groups):
+                raise ValueError("expected {} min_lrs, got {}".format(
+                    len(optimizer.param_groups), len(min_lr)))
+            self.min_lrs = list(min_lr)
```

Comment:
It won't make problems if min_lr is a list?

Reply:
I copied this part from Pytorch so I think it is ok